modules/jdbc/src/integrationTest/java/org/apache/ignite/jdbc/ItJdbcTransactionTest.java

```diff
@@ -0,0 +1,307 @@
+            checkUpdate(5, 10, (off, cnt) -> {
+                for (int i = off; i < off + cnt; i++) {
+                    pstmt.setInt(1, i);
+                    pstmt.setString(2, "name-" + i);
+
+                    pstmt.addBatch();
+                }
+
+                pstmt.executeBatch();
+            });
+        }
+    }
+
+    /**
+     * Ensures that data updates made in an explicit transaction can be rolled back and are
+     * not visible outside of an active transaction until they are committed.
+     *
+     * @param ops Number of operations.
+     * @param rowsPerOp Number of rows per operation.
+     * @param batchOp Update operation.
+     * @throws SQLException If failed.
+     */
+    private void checkUpdate(int ops, int rowsPerOp, BiConsumerX<Integer, Integer> batchOp) throws SQLException {
+        conn.setAutoCommit(false);
+
+        // Check rollback.
+        checkTxResult(ops, rowsPerOp, batchOp, false);
+
+        // Check commit.
+        checkTxResult(ops, rowsPerOp, batchOp, true);
+    }
+
+    /**
+     * Performs update operations and checks the result of committing or rolling back a transaction.
+     *
+     * @param ops Number of operations.
+     * @param rowsPerOp Number of rows per operation.
+     * @param batchOp Update operation.
+     * @param commit {@code True} to check transaction commit, {@code false} to check rollback.
+     * @throws SQLException If failed.
+     */
+    private void checkTxResult(int ops, int rowsPerOp, BiConsumerX<Integer, Integer> batchOp, boolean commit) throws SQLException {
+        // Creating another connection to ensure that the changes are not visible outside of the transaction.
+        try (Connection conn0 = DriverManager.getConnection(URL)) {
+            for (int i = 0; i < ops; i++) {
+                int off = rowsPerOp * i;
+
+                batchOp.accept(off, rowsPerOp);
+                assertEquals(off + rowsPerOp, rowsCount(conn));
+                assertEquals(0, rowsCount(conn0));
+            }
+
+            if (commit) {
+                conn.commit();
+            } else {
+                conn.rollback();
+            }
+
+            int expCnt = commit ? ops * rowsPerOp : 0;
+
+            assertEquals(expCnt, rowsCount(conn));
+            assertEquals(expCnt, rowsCount(conn0));
+        }
+    }
+
+    /**
+     * Gets the number of rows in the test table using specified connection.
+     *
+     * @param conn Active connection.
+     * @return Number of rows in the test table.
+     * @throws SQLException If failed.
+     */
+    private int rowsCount(Connection conn) throws SQLException {
+        try (Statement stmt = conn.createStatement()) {
+            try (ResultSet rs = stmt.executeQuery("select count(*) from TEST")) {
+                rs.next();
+
+                return rs.getInt(1);
+            }
+        }
+    }
+
+    @FunctionalInterface
+    private interface BiConsumerX<T, U> {
```

Comment:
Why not introduce an interface with a meaningful name?

Reply:
Thanks, fixed.